packages/statemanager/src/cache/account.ts

```diff
@@ -57,7 +57,7 @@ export class AccountCache extends Cache {
       } else {
         oldElem = this._orderedMapCache!.getElementByKey(cacheKeyHex)
       }
-      this._diffCache[this._checkpoints].set(cacheKeyHex, oldElem)
+      this._diffCache[this._checkpoints].set(cacheKeyHex, oldElem ?? { accountRLP: undefined })
```

Comment:
I would still think this is a fix "in the wrong direction", and I wonder why this is working now.

See the comment/differentiation in the code docs above:

>    * If the whole cache element is undefined (in contrast
>   * to the account), the element didn't exist in the cache
>  * before.

As I wrote in some explanations before, the diff cache should only be about the diff *in the cache*. We now artifically set here the information that the account does not exist (at all, so in the state) for the prestate.

We just do not know this (and also don't want to know this at this point). We are just reading the old cache element and if this returns `undefined` the pre state for the cache is just `undefined` as well, transporting the information that we do not know anything about this accout at this pre state point.

Reply:
I changed to a more intuitive fix, since the overwriting of the `undefined` value was the real issue at the heart of the problem.